stage5.3_fuller_application/pycasa/ui/image_folder_view.py

```diff
@@ -117,23 +117,25 @@ def _year_mask_default(self):
         return pd.Series(np.ones(len(self.model.data), dtype=bool))
 
     def _all_data_default(self):
+        # Enrich metadata with missing fields: date time, make
         data = self.model.data.copy()
+
         if DATETIME_COL not in data.columns:
             data[DATETIME_COL] = np.nan
 
         if MAKE_COL not in data.columns:
             data[MAKE_COL] = np.nan
 
+        def parse_year(x):
+            return x.split(":")[0] if isinstance(x, str) else "unknown"
+        data[YEAR_KEY] = data[DATETIME_COL].apply(parse_year)
+
```

Comment:
This isn't necessary to fix the bug but I accidentally included it in this commit. I'll let it be here because all missing columns are now added in one place so it's slightly easier to follow.